src/websockets/TimerService/index.js

```diff
@@ -1,328 +1,293 @@
+We pause the timer and check it's mode to set the time to 0 or the goal.
+Then we set the stopped flag to false.
+*/
+const clearTimer = (client) => {
+  pauseTimer(client);
+  client.time = client.countdown ? client.goal : 0;
+  client.stopped = false;
+};
 
-        // Prevent pausing application twice
-        if (!timerObject?.isRunning) {
-          return timerObject;
-        }
+/*
+Here we stop the timer.
+We pause the timer and set the stopped flag to true.
+*/
+const stopTimer = (client) => {
+  pauseTimer(client);
+  client.stopped = true;
+};
 
-        const currentTimeInSeconds = new Date().getTime() / 1000;
+/*
+Here we switch the timer mode.
+We pause the timer and check it's mode to set the time to 0 or the goal.
+*/
+const switchMode = (client) => {
+  client.countdown = !client.countdown;
+  client.time = client.countdown ? client.goal : 0;
+  client.paused = true;
+};
 
-        const totalTimeAddedUp = calculateTotalSecondsBasedOnStartingTime({
-          currentTimeInSeconds,
-          startedAtInSeconds: timerObject?.startedAtInSeconds,
-          seconds: timerObject?.seconds,
-        });
+// Here we get the goal time from the message.
+const getGoal = msg => parseInt(msg.split('=')[1]);
 
-        const newTimerObject = {
-          ...timerObject,
-          isRunning: false,
-          isUserPaused,
-          isApplicationPaused,
-          seconds: totalTimeAddedUp,
-        };
+// Here we set the goal and time to the goal time.
+const setGoal = (client, msg) => {
+  const goal = getGoal(msg);
+  client.goal = client.time = goal;
+};
 
-        /**
-         * Mongo Saving
-         *
-         * We would like to clean up here
-         * when we are saving persistent timer
-         * data. We only save when the user is
-         * logging out or all active connections have
-         * severed.
-         */
-        if (saveDataToDatabase) {
-          await setTimerToDatabase({
-            userId,
-            timerObject: {
-              // We use previous data since this is the correct time here
-              totalSeconds: totalTimeAddedUp,
-              ...newTimerObject,
-            },
-          });
+const goalOver10Hours = (client, time) => {
+  const goal = moment.duration(client.goal).add(time, 'milliseconds').asHours();
+  return goal > 10;
+};
 
-          await deleteTimerFromRedis({
-            userId,
-            redisClient: main,
-          });
-        } else {
-          // Resync timer to redis
-          await saveTimerToRedis({
-            userId,
-            redisClient: main,
-            timerObject: newTimerObject,
-          });
+/*
+Here we add the goal time.
+First we get the goal time from the message.
+Then we add it to the current goal time and set it as the new goal time.
+We also add it to the current time and set it as the new time.
+*/
+const addGoal = (client, msg) => {
+  const goal = getGoal(msg);
+  if (goalOver10Hours(client, goal)) return;
+
```

Comment:
```suggestion

  if (!client.paused) {
    client.time = getTotalElapsedTime(client).asMilliseconds().toFixed();
    client.lastAccess = moment();
  }
  
```
Adding or removing time from the goal while the timer is running causes the timer to show an incorrect time for a second because client.time is only updated on pauses. This appears to fix it.